### PR TITLE
feat(typeahead): Adds 'select on exact' setting.

### PR DESCRIPTION
**Pull request by VoltiSubito**
While using the typeahead directive, we desired that it automatically select an item when the user input exactly matches the option in the dropdown.  This PR is an implementation of this feature.

I called the new attribute typeahead-select-on-exact and it takes a boolean.  I've added tests and updated the docs.  I tested it locally against IE 10, FF 32, and Chrome 41 using Angular 1.3.13 and 1.2.26.

I also believe this will resolve issue #3310

### CODE REVIEW

**Comment by cssensei** [comment]
:+1:

**Comment by mrmojorisin0625** [comment]
:+1:

**Comment by abondin** [comment]
:+1:

File: src/typeahead/docs/readme.md [hunk 1]
```diff
@@ -27,6 +27,10 @@ The typeahead directives provide several attributes:
 * `typeahead-editable` <i class="glyphicon glyphicon-eye-open"></i>
    _(Defaults: true)_ :
    Should it restrict model values to the ones selected from the popup only ?
+   
+* `typeahead-focus-first`
+   _(Defaults: true)_ :
+   Should the first match automatically be focused as you type?
```
Comment (wesleycho) [inline src/typeahead/docs/readme.md]: Any reason this was moved?

File: src/typeahead/test/typeahead.spec.js [hunk 2]
```diff
@@ -389,6 +389,60 @@ describe('typeahead tests', function () {
       expect($scope.result).toEqual('AL');
       expect(inputEl.val()).toEqual('AL');
     });
+    
+    it('should select on an exact match when set', function () {
```
Comment (wesleycho) [inline src/typeahead/test/typeahead.spec.js]: All of these should be grouped under a describe block with the select on an exact match descriptor string.

**Comment by wesleycho** [comment]
I like this overall - I am going to schedule this for 0.13.1, since we locked down 0.13.0 for Angular 1.3 compatibility.

File: src/typeahead/docs/readme.md (hunk 1 above)
Comment (VoltiSubito) [inline src/typeahead/docs/readme.md]: When I was looking for where to add the documentation for the new attribute, I noticed that they were almost in alphabetical order (except for focus-first) and assumed that they were supposed to be so I moved 'focus-first' to higher up in the list.  Looking again, I see I misplaced 'select-on-exact' anyway... so I've fixed that now.  I'm happy to order them differently if I'm incorrect.

File: src/typeahead/test/typeahead.spec.js (hunk 2 above)
Comment (VoltiSubito) [inline src/typeahead/test/typeahead.spec.js]: Yes I see I missed that.  Done.

**Comment by VoltiSubito** [comment]
@wesleycho I want to squash this but don't want to destroy history for future reviewer(s). Would you mind if I squash this soon(ish), or should I wait until after it has been reviewed?

File: src/typeahead/typeahead.js [hunk 3]
```diff
@@ -154,6 +166,11 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position', 'ui.bootstrap
               scope.position.top = scope.position.top + element.prop('offsetHeight');
 
               element.attr('aria-expanded', true);
+              
+              //Select the single remaining option if user input matches
+              if(selectOnExact && scope.matches.length === 1 && inputIsExactMatch(inputValue, 0)){
```
Comment (wesleycho) [inline src/typeahead/typeahead.js]: Add a space here after `if`

File: src/typeahead/typeahead.js [hunk 4]
```diff
@@ -120,6 +123,15 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position', 'ui.bootstrap
           element.attr('aria-activedescendant', getMatchId(index));
         }
       });
+      
+      var inputIsExactMatch = function(inputValue, index) {
+        
+        if(scope.matches.length > index && inputValue){
```
Comment (wesleycho) [inline src/typeahead/typeahead.js]: Add a space after `if`

**Comment by wesleycho** [comment]
I'm fine with squashing now - the merge commits are polluting the history, and would be troublesome to rebase with.

**Comment by VoltiSubito** [comment]
Squashed.

**Pull request closed without merging**